testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -313,70 +318,272 @@ impl RewardSetEventPayload {
     }
 }
 
+/// Convert a NetError into an io::Error if appropriate.
+fn handle_net_error(e: NetError, msg: &str) -> io::Error {
+    if let NetError::ReadError(ioe) = e {
+        ioe
+    } else if let NetError::WriteError(ioe) = e {
+        ioe
+    } else if let NetError::RecvTimeout = e {
+        io::Error::new(io::ErrorKind::WouldBlock, "recv timeout")
+    } else {
+        io::Error::new(io::ErrorKind::Other, format!("{}: {:?}", &e, msg).as_str())
+    }
```

Comment:
`match` would be cleaner:

```suggestion
    match e {
        NetError::ReadError(ioe) => ioe,
        NetError::WriteError(ioe) => ioe,
        NetError::RecvTimeout = io::Error::new(io::ErrorKind::WouldBlock, "recv timeout"),
        _ => io::Error::new(io::ErrorKind::Other, &format!("{e}: {msg:?}")),
    }
```

Reply:
Sure